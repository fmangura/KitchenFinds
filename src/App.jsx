import React, {useState} from 'react'
import { BrowserRouter, Routes, Route } from 'react-router-dom';

import Auth from './components/AuthProvider'
import Home from './routes/Home'
import LoginForm from './routes/LoginForm'
import RegisterForm from './routes/RegisterForm'
import Nav from './components/Nav';
import Profile from './routes/Profile'

export default function App () {

    return (
        <>
            <BrowserRouter>
                <Auth>
                    <Nav />
                    <Routes>
                        <Route path='/' element={<Home />} />
                        <Route path='/profile' element={<Profile />} />
                        <Route path='/login' element={<LoginForm />} />
                        <Route path='/register' element={<RegisterForm />} />
                    </Routes>
                </Auth>
            </BrowserRouter>
        </>
    );
}