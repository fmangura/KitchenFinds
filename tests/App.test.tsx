import React from 'react';
import {getByRole, render, screen} from '@testing-library/react'
import App from '../src/App'
import { test } from 'vitest';

describe('App', () => {
    it('renders without crashing', () => {
        render(<App />)   
    });

    it('Snapshot', () => {
        const { asFragment } = render(<App />);
        expect(asFragment).toMatchSnapshot();
    });

    it('Proper elements render', () => {
        render(<App />);
        screen.getByRole('button', {name:'Login'})
    })
})