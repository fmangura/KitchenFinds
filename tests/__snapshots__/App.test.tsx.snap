// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Snapshot 1`] = `[Function]`;
